carve.lic

```diff
@@ -0,0 +1,118 @@
+=begin
+  Suggestions and contributions are welcome: https://github.com/rpherbig/dr-scripts
+=end
+
+custom_require(%w(common common-crafting))
+
+class Carve
+  include DRC
+  include DRCC
+
+  def initialize(args:[])
+    settings = get_settings
+    @bag = settings.crafting_container('backpack')
+    @finish, @type, @chapter, @recipe_name, @material, @noun = args
+
+    case @type
+    when /carving/
+      carve_item
+    else
+      echo('Carve does not yet support that form of carving.')
+    end
+  end
+
+  def get_or_fail(item)
+    waitrt?
+    case bput("get my #{item}", '^You get', '^You are already', '^You pick up', '^What do you', '^What were you')
+    when 'What do you', 'What were you'
+      echo("You seem to be missing: #{item}")
+      exit
+    end
+  end
+
+  def store_thing(item)
+    waitrt?
+    bput("put my #{item} in my #{@bag}", 'You put your', 'What were you referring to')
+  end
+
+  def turn_to(section)
+    bput("turn my book to #{section}", 'You turn your', 'The book is already')
+  end
+
+  def carve_item
+    get_or_fail('carving book')
+    turn_to("chapter #{@chapter}")
+    turn_to("page #{find_recipe(@chapter, @recipe_name)}")
+    bput('study my book', 'Roundtime')
+    store_thing('book')
+    get_or_fail('chisel')
+    get_or_fail("#{@material} rock")
+    carve('cut my rock with my chisel')
+  end
+
+  def carve(command)
+    waitrt?
+    case bput(command,
+              'rough, jagged edges',
+              'determine it is no longer level',
+              'developed an uneven texture along its surface',
+              #'Applying the final touches',
```

Comment:
Regarding polish, adding a quick get/analyze on it should work.  I don't know that I've ever used it more than once on a given item, but like the rifflers/rasp, I'm sure it's possible.   Polish cannot be COMBINEd.  Should polish with less than 3-5 uses be discarded in order to combat possible hoarding due to invendory reversal at login?